### ISSUE TITLE
TlsClientProtocol adds ExtendedMasterSecretExtension unconditionally in ClientHello message

### ISSUE THREAD

**Issue by onyxbits** [bcgit/bc-java#654]
I just tried upgrading from bctls-jdk15on 1.57 to 1.64 and found the newer version no longer working for me as it always adds an extended master secret extension to the client hello message. Can this be made configurable? 

My use case is about dealing with SSL fingerprinting, which means, I need exact control over every byte in the hello message.

**Comment by peterdettman** [comment]
Yes, I think we can make this configurable; we need it to support the system property jdk.tls.useExtendedMasterSecret in BCJSSE anyway.

**Comment by peterdettman** [comment]
We've added support for jdk.tls.useExtendedMasterSecret in BCJSSE, and in the low-level TLS API one can override shouldUseExtendedMasterSecret() in your client/server objects.

There's a new beta version 165b08 up at https://downloads.bouncycastle.org/betas/ with the changes if you would like to test it.

**Comment by kimmerin** [comment]
Is it possible to have an influence on that setting when using JSSE and not the low level API. I'd really like to change that setting on a per-connection level rather than disabling that globally because of just one server out there that doesn't support it.

**Comment by peterdettman** [comment]
Currently we've just followed SunJSSE, and I don't think there's currently any way to do that.

We're contemplating a BCJSSE-specific extension that might allow per-SSLContext and possibly per-connection overrides of all system properties, which would cover this case. Does that sound like what you have in mind?

**Comment by kimmerin** [comment]
Per-Context and per-connection is completely OK for me. If SSL-handshakes take place in the same thread that is calling startHandshake, retrieving the System-Property by using org.bouncycastle.util.Properties could be an immediate solution that doesn't break any existing frameworks.
